### PR TITLE
Metadata Discriminators

### PR DESCRIPTION
**Pull request by ieaves**

### CODE REVIEW

File: grai-cli/grai_cli/api/server/endpoints.py [hunk 1]
```diff
@@ -26,11 +26,19 @@ def is_authenticated():
...
     to_file: Optional[Path] = typer.Option(None, "--f", help="Write nodes to file"),
 ):
     client = get_default_client()
-    result = client.get("Node")
+    if name is None:
+        result = client.get("Node")
+    elif namespace is None:
+        result = client.get("Node", name)
+    else:
+        utilities.print(name, namespace)
```
Comment (edlouth) [inline grai-cli/grai_cli/api/server/endpoints.py:40]: We can probably remove this line

**Review by edlouth — APPROVED** [review]

File: grai-cli/grai_cli/api/server/endpoints.py (hunk 1 above)
Comment (edlouth) [reply grai-cli/grai_cli/api/server/endpoints.py:40]: `utilities.print(name, namespace)`

File: grai-server/app/connections/task_helpers.py [hunk 2]
```diff
@@ -48,27 +47,32 @@ def update(
...
             if type == "Edge":
-                spec["source"] = NodeID(**active_model.source.__dict__)
-                spec["destination"] = NodeID(**active_model.destination.__dict__)
+                spec["source"] = NodeNamedID(**active_model.source.__dict__)
+                spec["destination"] = NodeNamedID(**active_model.destination.__dict__)
+                spec["metadata"] = {
+                    "grai": {"node_type": "Edge", "edge_type": "ColumnToColumn"}
+                }
+            else:
+                spec["metadata"] = {"grai": {"node_type": "Column"}}
```
Comment (edlouth) [inline grai-server/app/connections/task_helpers.py:56]: Not at all convinced by these lines, could you take a look @ieaves

Comment (ieaves) [reply grai-server/app/connections/task_helpers.py]: Not using the schemas ended up being potentially just as bad but so it goes.

**Pull request merged**